File: app/src/main/kotlin/com/moviereel/di/components/ActivityComponent.java
package com.moviereel.di.components;

import com.moviereel.di.PerActivity;
import com.moviereel.di.modules.ActivityModule;
import com.moviereel.ui.activities.customerror.CustomErrorActivity;
import com.moviereel.ui.activities.intro.AppIntroduction;
import com.moviereel.ui.activities.intro.splash.SplashActivity;
import com.moviereel.ui.activities.itemdetails.movie.MovieDetailsActivity;

import com.moviereel.ui.activities.main.MainActivity;
import com.moviereel.ui.activities.settings.SettingsActivity;
import com.moviereel.ui.fragments.movie.nowplaying.MovieNPFragment;

import dagger.Component;

/**
 * @author lusinabrian on 27/03/17
 */
@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {

    void inject(SplashActivity splashActivity);

    void inject(CustomErrorActivity customErrorActivity);

    void inject(AppIntroduction appIntroduction);

    void inject(MainActivity mainActivity);

    void inject(SettingsActivity settingsActivity);

    void inject(MovieNPFragment movieNPFragment);

    void inject(MovieDetailsActivity movieDetails);
}

File: app/src/main/kotlin/com/moviereel/di/components/ActivityComponent.kt
package com.moviereel.di.components

import com.moviereel.di.PerActivity
import com.moviereel.di.modules.ActivityModule
import com.moviereel.ui.customerror.CustomErrorActivity
import com.moviereel.ui.intro.AppIntroduction
import com.moviereel.ui.intro.splash.SplashActivity
import com.moviereel.ui.movie.MovieDetailsActivity

import com.moviereel.ui.main.MainActivity
import com.moviereel.ui.settings.SettingsActivity
import com.moviereel.ui.movie.nowplaying.MovieNPFragment

import dagger.Component

/**
 * @author lusinabrian on 27/03/17
 */
@PerActivity
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(ActivityModule::class))
interface ActivityComponent {

    fun inject(splashActivity: SplashActivity)

    fun inject(customErrorActivity: CustomErrorActivity)

    fun inject(appIntroduction: AppIntroduction)

    fun inject(mainActivity: MainActivity)

    fun inject(settingsActivity: SettingsActivity)

    fun inject(movieNPFragment: MovieNPFragment)

    fun inject(movieDetails: MovieDetailsActivity)
}
